lib/job.js

```diff
@@ -204,21 +204,26 @@ Job.prototype.promote = function(){
     });
 };
 
+/**
+ * Attempts to retry the job. Only a job that has failed can be retried.
+ *
+ * @return {Promise} If resolved and return code is 1, then the queue emits a waiting event
+ * otherwise the operation was not a success and throw the corresponding error. If the promise
+ * rejects, it indicates that the script failed to execute
+ */
 Job.prototype.retry = function(){
```

Comment:
Cool, I made the script accept an options parameter to determine on which state the job should be when attempting to reprocess it. This should support retrying a job (from a failed state) or repeating a job (from the completed state).

As for the delay, I took a dive into it, but it was taking me too much time, maybe that can be added on a separate PR?